### PR TITLE
Worked on coding concepts, interview questions, and some of the codin…

### PR DESCRIPTION
**Pull request by SpiralCode28**
Worked on coding concepts, interview questions, and some of the coding challenges. Need help with the latter.

### CODE REVIEW

**Comment by SpiralCode28** [comment]
I have completed the assessment. :)

File: code-challenges.js [hunk 1]
```diff
@@ -5,35 +5,151 @@
...
+// ^^^ variable one
+
 var myNumbers2 = [8, -7, 42, 9, 13]
+// ^^^ variable two
+
+var allMyNumbers = myNumbers1.concat(myNumbers2);
+// ^^^ variable 3 - using concat() to combine two arrays
+
+ console.log(allMyNumbers.length);
+ // .length to log the length of the array
```
Comment (sjproctor) [inline code-challenges.js:76]: Nice job!

File: code-challenges.js [hunk 2]
```diff
@@ -5,35 +5,151 @@
...
+    // log the joined array that became a string
+    console.log(joinArray);
+
+    // Return the reversed string
+    return joinArray;
+
+}
+
+// Call on function above
+reverseString("Echo 2021");
```
Comment (sjproctor) [inline code-challenges.js:114]: I'm glad functions helped you! It looks like what is in the function is exactly what you need for this challenge.

File: code-challenges.js [hunk 3]
```diff
@@ -5,35 +5,151 @@
...
+for (let i = 0; i < myArray.length; i++) //for loop
+
+if (myArray[i] % 2 ===0 ) {
+  console.log("odd");
+  // ^^^ will log if odd
+
+} else {
+  console.log("even");
+  // ^^^ will log if even
+}
```
Comment (sjproctor) [inline code-challenges.js:131]: Nice!

File: code-challenges.js [hunk 4]
```diff
@@ -5,35 +5,151 @@
...
+  console.log(number2 - number1);
+  //  ^^^ log equation if if statement is true
+
+} else { // condition 2
+  // ^^^ else runs only if the condition above was not met
+
+  console.log(number1 - number2);
+  //  ^^^ log equation if else statement is true
 
+}
```
Comment (sjproctor) [inline code-challenges.js:153]: Looks great!

File: coding-concepts.js [hunk 5]
```diff
@@ -8,44 +8,58 @@
...
 
-// a) Your answer:
+// a) Your answer: SATURDAYSUNDAY
 // b) Verify and explain:
-
+  // The answer above is not correct.
+  // This is what is logged: weekendDays.toUpperCase is not a function
+  // When the array is assigned by the variable weekendDays, The computer doesn't know what to log into UpperCase
+  // This could be changed if the array was a string "saturday and sunday" without the square brackets or comma.
+    // It would then log = SATURDAY AND SUNDAY
```
Comment (sjproctor) [inline coding-concepts.js:53]: Yup, this gives an error. It is an important error to note. `.toUpperCase` is a method that will only work on strings.

File: coding-concepts.js [hunk 6]
```diff
@@ -8,44 +8,58 @@
...
-// console.log(typeof dataTypes.length)
+  console.log(typeof dataTypes.length)
 
-// a) Your answer:
+// a) Your answer: number
 // b) Verify and explain:
+  // The answer above is correct.
+  // typeof is a way you can check the type of a variable
+  // The code is telling the computer to log "typeof" (out of - number, string, or object) of the variable "dataTypes" length
+  // We know that the length of the dataTypes variable will be a number.
```
Comment (sjproctor) [inline coding-concepts.js:65]: Nice!

File: interview-questions.md [hunk 7]
```diff
@@ -9,58 +9,87 @@ Then, research the question to expand on your answer. Even if you feel you have
...
   Researched answer:
 
+  - GitHub is an online platform and graphical user interface that works for the local, git software. Git is software that is local and not online.
 
 
 2. What types of evaluations will return a Boolean value?
 
   Your answer:
 
+  - True or false questions
```
Comment (sjproctor) [inline interview-questions.md:23]: Sure, but when can you expect to see these? Specifically we are looking for logical, equality, and relational operators.

File: interview-questions.md [hunk 8]
```diff
@@ -9,58 +9,87 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+    - keep distractions at a minimum
+    - both have eyes on the code
+
 
 
 6. What is something we did in class this week you found helpful?  
 
   Your answer:
 
+  - I'm a visual leaner and I find it was VERY helpful when Austin drew out (on a whiteboard) his explanation of what the GitHub's pulling and pushing process looked like. He also showed how multiple people - on one project - work from "main" on their separate branches.
```
Comment (sjproctor) [inline interview-questions.md:78]: Yay that is great!

**Review by sjproctor — APPROVED** [review]
Nice work Hannah! You seem to be processing the information really well. Keep using your resources and balancing your time on each section.

**Pull request merged**